src/bin/bench.rs

```diff
@@ -185,19 +188,40 @@ fn perform_tree_operations(tree: Tree, config: Config) {
         (Op::Delete, config.delete),
         (Op::Cas, config.cas),
     ];
-
     // TODO update to handle division errors
     let ops_per_thread = config.num_operations / config.num_threads;
     let mut threads = Vec::new();
+    let statistics = Statistics::new();
+    let (tx, rx) = channel();
+
+    let statistics = Arc::new(statistics);
+    thread::spawn(move || {
+        loop {
+            // TODO how do I get the most recent value here rather than receiving one at a time?
+            let ops_per_second = rx.recv().unwrap();
+
+            println!("{:?}", ops_per_second);
+
+            // TODO how can I mutably borrow the immutable field ops_per_second
+            statistics.ops_per_second.push(ops_per_second);
+            thread::sleep(Duration::from_secs(1));
+        }
+    });
+
     let tree = Arc::new(tree);
     let ops = Arc::new(ops);
-
+    let ops_per_second = Arc::new(AtomicUsize::new(0));
     for i in 0..config.num_threads {
         let tree = tree.clone();
         let ops = ops.clone();
+        let ops_per_second = ops_per_second.clone();
+        let tx = tx.clone();
         let t = thread::spawn(move || for _ in 0..ops_per_thread {
             let op_to_perform = get_operation_choice(&ops, sum_ops);
 
+            ops_per_second.fetch_add(1, Ordering::SeqCst);
+            tx.send(ops_per_second.load(Ordering::SeqCst));
```

Comment:
for any of these counters, we can use the `Relaxed` memory ordering, which is still atomic, but does not slow down other threads so much.